processor.js

```diff
@@ -16,55 +16,74 @@ var notifier_messager = require("./messager/messager")({
   queueUrl: process.env.OUTGOING_QUEUE_URL
 });
 
-var mailroom = require('webmaker-mailroom')();
+var rateLimitConfig,
+    ttl = process.env.TTL || 1000 * 60 * 5;
 
-var workers = async.applyEachSeries([
-  worker.archiver(archiver_config),
-  worker.backwards_compatibility,
-  worker.remind_user_about_event(notifier_messager, mailroom),
-  worker.login_request(notifier_messager, mailroom),
-  worker.send_sms(notifier_messager),
-  worker.receive_coinbase_donation(notifier_messager, mailroom),
-  worker.reset_request(notifier_messager, mailroom),
-  worker.send_event_host_email(notifier_messager, mailroom),
-  worker.send_mofo_staff_email(notifier_messager, mailroom, process.env.MOFO_STAFF_EMAIL),
-  worker.send_new_user_email(notifier_messager, mailroom),
-  worker.event_mentor_confirmation_email(notifier_messager, mailroom),
-  worker.event_coorganizer_added(notifier_messager, mailroom),
-  worker.sign_up_for_bsd(notifier_messager),
-  worker.badge_awarded_send_email(notifier_messager),
-  worker.badge_application_denied(notifier_messager, mailroom),
-  worker.hive_badge_awarded(notifier_messager, mailroom),
-  worker.suggest_featured_resource(notifier_messager, process.env.SFR_SPREADSHEET, process.env.SFR_WORKSHEET)
-]);
+if ( process.env.REDIS_HOST && process.env.REDIS_PORT ) {
+  rateLimitConfig = {
+    host: process.env.REDIS_HOST,
+    port: process.env.REDIS_PORT,
+    database: process.env.REDIS_DATABASE || 0,
+    password: process.env.REDIS_AUTH,
+    partition: "sawmill"
+  };
+}
 
-var SqsQueueParallel = require('sqs-queue-parallel');
-var config = {
-  accessKeyId: process.env.AWS_ACCESS_KEY_ID,
-  secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
-  name: process.env.INCOMING_QUEUE_NAME,
-  region: process.env.AWS_QUEUE_REGION,
-  debug: process.env.DEBUG
-};
-var queue = new SqsQueueParallel(config);
+require("./rate-limit")(rateLimitConfig, ttl, startProcessor);
+
+
+function startProcessor(rateLimitClient) {
 
-queue.on("message", function(m) {
-  workers(m.message.MessageId, m.data, function(err) {
-    if (err) {
-      console.log(err);
-      return m.next();
-    }
+  var mailroom = require('webmaker-mailroom')();
 
-    if (config.debug) {
-      console.log('SAWMILL EVENT [' + m.data.event_type + ']: %j', m.data.data);
-    }
+  var workers = async.applyEachSeries([
```

Comment:
I think we can leave this where it was, because once we've created the cache, we can just pass it in as an option